third-party/linshare/src/main/java/org/apache/james/linshare/client/LinshareAPI.java

```diff
@@ -112,4 +116,12 @@ default void deleteAllDocuments() {
     @RequestLine("GET /linshare/webservice/rest/user/v2/received_shares")
     @feign.Headers({ ACCEPT_APPLICATION_JSON, CONTENT_TYPE_APPLICATION_JSON })
     List<ReceivedShare> receivedShares();
+
+    @RequestLine("GET /linshare/webservice/rest/user/v2/received_shares/{documentId}/download")
+    @feign.Headers({ CONTENT_TYPE_APPLICATION_OCTET_STREAM, CONTENT_DISPOSITION_ATTACHMENT })
+    byte[] downloadShare(@Param("documentId") String documentId, @Param("filename") String filename);
+
+    default byte[] downloadShare(Document.DocumentId documentId, String filename) {
```

Comment:
That is maybe linshare API that is crappy, but why is filename needed at all?

Reply:
More like `Feign`... Had to add the header here : 
```
String CONTENT_DISPOSITION_ATTACHMENT = "Content-Disposition: attachment; filename=\"{filename}\"";
```
Somehow it doesn't seem to understand there is a attachment to download and fails without this header...

Reply:
Thinking about this - this endpoint is only used for testing - and have some impact on other endpoints (decoder). 

I would rather ship it as part of the LinshareAPIForTesting